geth-utils/gethutil/mpt/trie/stacktrie.go

```diff
@@ -701,33 +695,17 @@ func (st *StackTrie) GetProof(db ethdb.KeyValueReader, key []byte) ([][]byte, er
 			isHashed = true
 			c_rlp, error := db.Get(c.val)
 			if error != nil {
-				fmt.Println(error)
 				panic(error)
 			}
-			fmt.Println(c_rlp)
 
 			proof = append(proof, c_rlp)
+			branchChild := st.getNodeFromBranchRLP(c_rlp, k[i])
 
-			for i < len(k)-1 {
-				node := st.getNodeFromBranchRLP(c_rlp, k[i])
-				i += 1
-				fmt.Println(node)
-
-				if len(node) == 1 && node[0] == 128 { // no child at this position
-					break
-				}
-
-				c_rlp, error = db.Get(node)
-				if error != nil {
-					fmt.Println(error)
-					panic(error)
-				}
-				fmt.Println(c_rlp)
-
-				proof = append(proof, c_rlp)
+			if len(branchChild) == 1 {
+				// no child at this position - 128 is RLP encoding for nil object
```

Comment:
question, you said `128 is RLP encoding for nil object`, but you didn't do any checks like your previous code (`node[0] == 128`). So, do we need to add `node[0] == 128`? Or `len(branchChild)` is enough to represent that there is no child for this branch?

Reply:
I excluded `node[0] == 128` because the break is needed also in the case when `c_rlp` is a transaction leaf (and not a branch or extension node). Would be good to have some more tests to verify this behaviour though. I added a comment here https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1779/commits/16d93ad655487b85d115401219dfd3f4d9010ccf